### PR TITLE
feat: scopes are supported in all the OAuth2 flows

### PR DESCRIPTION
**Pull request by XVincentX**
If adding a new definition:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: types/swagger-schema-official/index.d.ts [hunk 1]
```diff
@@ -182,40 +182,38 @@ export interface BasicAuthenticationSecurity extends BaseSecurity {
 export interface ApiKeySecurity extends BaseSecurity {
   type: 'apiKey';
   name: string;
-  in: string;
+  in: 'query' | 'header';
```
Comment (mohsen1) [inline types/swagger-schema-official/index.d.ts]: do you have a link to specs that verifies this?

Comment (XVincentX) [reply types/swagger-schema-official/index.d.ts]: Sure! https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#security-definitions-objec

---

image

Comment (XVincentX) [reply types/swagger-schema-official/index.d.ts]: @mohsen1 Any feedback?

**Comment by XVincentX** [comment]
@RyanCavanaugh Any chance we can merge this? It's being here for a while now 👍

**Comment by XVincentX** [comment]
@mohsen1 Any chance you can have another look at this? I've been implementing different fixes that would really be beneficial for anybody working with this stuff.

Thanks!

**Review by mohsen1 — APPROVED** [review]
LGTM

**Pull request merged**